src/e2ee/matrixKeyProvider.test.ts

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { describe, expect, test, vi } from "vitest";
+import {
+  MatrixRTCSession,
+  MatrixRTCSessionEvent,
+} from "matrix-js-sdk/src/matrixrtc";
+import { KeyProviderEvent } from "livekit-client";
+
+import { MatrixKeyProvider } from "./matrixKeyProvider";
+
+function mockRTCSession(): MatrixRTCSession {
+  return {
+    on: vi.fn(),
+    off: vi.fn(),
+    reemitEncryptionKeys: vi.fn(),
+  } as unknown as MatrixRTCSession;
+}
+
+describe("matrixKeyProvider", () => {
+  test("initializes", () => {
+    const keyProvider = new MatrixKeyProvider();
+    expect(keyProvider).toBeTruthy();
+  });
+
+  test("listens for key requests and emits existing keys", () => {
+    const keyProvider = new MatrixKeyProvider();
+
+    const session = mockRTCSession();
+
+    keyProvider.setRTCSession(session);
+
+    expect(session.on).toHaveBeenCalledWith(
+      MatrixRTCSessionEvent.EncryptionKeyChanged,
+      expect.any(Function),
+    );
+    expect(session.off).not.toHaveBeenCalled();
+  });
```

Comment:
I suppose we test that updating session (join/leave/receive new keys) also trigger a session `EncryptionKeyChanged` emit somewhere else?